prow/gerrit/adapter/adapter.go

```diff
@@ -464,17 +445,21 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 	switch change.Status {
 	case client.Merged:
 		var postsubmits []config.Postsubmit
-		for attempt := 0; attempt < inRepoConfigRetries; attempt++ {
-			postsubmits, err = c.inRepoConfigCacheHandler.GetPostsubmits(trimmedHostPath, func() (string, error) { return baseSHA, nil }, func() (string, error) { return change.CurrentRevision, nil })
-			// Break if there was no error, or if there was a merge conflict
-			if err == nil || strings.Contains(err.Error(), "Merge conflict in") {
-				break
+		if c.config().InRepoConfigEnabled(cloneURI) {
+			for attempt := 0; attempt < inRepoConfigRetries; attempt++ {
+				postsubmits, err = c.inRepoConfigCacheHandler.GetPostsubmits(cloneURI, func() (string, error) { return baseSHA, nil }, func() (string, error) { return change.CurrentRevision, nil })
+				// Break if there was no error, or if there was a merge conflict
+				if err == nil || strings.Contains(err.Error(), "Merge conflict in") {
+					break
+				}
 			}
+			if err := c.handleInRepoConfigError(err, instance, change); err != nil {
+				return err
+			}
+		} else {
+			postsubmits = c.config().PostsubmitsStatic[cloneURI]
 		}
-		if err := c.handleInRepoConfigError(err, instance, change); err != nil {
-			return err
-		}
-		postsubmits = append(postsubmits, c.config().PostsubmitsStatic[cloneURI.String()]...)
```

Comment:
We are now setting `postsubmits` to the static postsubmits rather than appending. I'm not sure why the original implementation was correct, doesn't `GetPostsubmits` include the static postsubmits? I would have expected the previous implementation to trigger duplicate jobs, but that wasn't happening?

Reply:
There is a duplication problem in existing implementation, decided to move the fix to https://github.com/kubernetes/test-infra/pull/27616 as it's orthogonal to current PR.

Reply:
Took me a while to understand what's going on here. The bug here was masked by the fact that `c.inRepoConfigCacheHandler.GetPostsubmits(trimmedHostPath` passed in trimmedHostPath, but static postsubmit jobs expect `https://` prefix, so it returned only inrepoconfig jobs. In contrast, `c.config().PostsubmitsStatic[cloneURI.String()]` contains `https://` prefix, which is why the final postsubmits still have static jobs. @cjwagner please review https://github.com/kubernetes/test-infra/pull/27616

Reply:
Gotcha, I think I remember this now. I'm not sure we can fix this in a separate PR though. I left a comment explaining on the other PR.

Reply:
make sense, I'll merge these two PRs

Reply:
merged